### ISSUE TITLE
data explorer jerks around during auto-complete because of scroll bar

### ISSUE THREAD

**Issue by mlucy** [rethinkdb/rethinkdb#3870]
When I open the data explorer on my laptop and start typing a query, the box jerks left and right because some of the auto-complete boxes are long enough that they cause the scroll bar to appear.  (@mglukhovsky @deontologician)

**Comment by mglukhovsky** [comment]
What browser / OS are you using?

Can you include a screenshot with an example where this happens?

We may want to truncate descriptions and use `overflow:hidden` for unusually long terms.

**Comment by mlucy** [comment]
Chrome, OS X.
![screen shot 2015-03-03 at 4 04 46 pm 2](https://cloud.githubusercontent.com/assets/1777134/6475450/32544d42-c1bf-11e4-9efe-53f055c8c3d1.png)
![screen shot 2015-03-03 at 4 04 50 pm 2](https://cloud.githubusercontent.com/assets/1777134/6475451/35454dc6-c1bf-11e4-97ae-d1b7957c9873.png)

**Comment by mlucy** [comment]
Maybe we could just have more columns?

**Comment by mglukhovsky** [comment]
@mlucy, when you say 'box', you're referring to the suggestion box, not the Data Explorer itself, right?

I agree that making a fourth column might be a good solution.

**Comment by mlucy** [comment]
No, I meant that the box you type into jerks left and right because the suggestion box causes the scroll bar to appear.

**Comment by mglukhovsky** [comment]
@mlucy and I discussed this in person. Options to consider:

- Add a scrollbar regardless of the amount of vertical content across the UI -- this is subpar on some browsers, where the scrollbar is visually unappealing, unlike OS X.
- Add a fourth column -- this may make the suggestion box too large.
- Compact the visual space each term takes up -- there's a lot of horizontal and vertical space padding.
- Move the syntax description box to the left or right of the suggestion box, rather than below -- we're have to detect how much horizontal space is available on either side to choose where to display it, and would cause a problem on small displays.
- Add some content below the Data Explorer, maybe explaining how to use the Data Explorer. This issue only shows up when you first load the DE and haven't entered a query.

I'd like to think about a few other options for this.

**Comment by marshall007** [comment]
If you're looking for a simple solution, I think this can be fixed by simply setting `overflow: hidden` on the `#sticky-wrap` element. The result is that the autocompletion just gets cut off at the bottom instead of changing the height of the page.

**Comment by mglukhovsky** [comment]
@marshall007: that solution makes sense. :smiley:

**Comment by thenightwassave** [comment]
That would make the content at the bottom hidden from view. Simple fix, yes; but I think it would be better to find a way to show all the information.

**Comment by danielmewes** [comment]
@thenightwassave I think that would be ok in this case since it only affects the suggestions for tab-completion. We can still try out different numbers of columns, compacting the space etc. to make more of them fit on the screen.
